github/resource_github_branch.go

```diff
@@ -0,0 +1,202 @@
+		return fmt.Errorf("Error creating GitHub branch reference %s/%s (%s): %s",
+			orgName, repoName, branchRefName, err)
+	}
+
+	d.SetId(buildTwoPartID(&repoName, &branchName))
+
+	return resourceGithubBranchRead(d, meta)
+}
+
+func resourceGithubBranchRead(d *schema.ResourceData, meta interface{}) error {
+	err := checkOrganization(meta)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+	if !d.IsNewResource() {
+		ctx = context.WithValue(ctx, ctxEtag, d.Get("etag").(string))
+	}
+
+	client := meta.(*Organization).client
+	orgName := meta.(*Organization).name
+	repoName, branchName, err := parseTwoPartID(d.Id(), "repository", "branch")
+	if err != nil {
+		return err
+	}
+	branchRefName := "refs/heads/" + branchName
+
+	log.Printf("[DEBUG] Querying GitHub branch reference %s/%s (%s)",
+		orgName, repoName, branchRefName)
+	ref, resp, err := client.Git.GetRef(ctx, orgName, repoName, branchRefName)
+	if err != nil {
+		if ghErr, ok := err.(*github.ErrorResponse); ok {
+			if ghErr.Response.StatusCode == http.StatusNotModified {
+				return nil
+			}
+			if ghErr.Response.StatusCode == http.StatusNotFound {
+				log.Printf("[WARN] Removing branch %s/%s (%s) from state because it no longer exists in Github",
+					orgName, repoName, branchName)
+				d.SetId("")
+				return nil
+			}
+		}
+		return fmt.Errorf("Error querying GitHub branch reference %s/%s (%s): %s",
+			orgName, repoName, branchRefName, err)
+	}
+
+	d.SetId(buildTwoPartID(&repoName, &branchName))
+	d.Set("etag", resp.Header.Get("ETag"))
+	d.Set("repository", repoName)
+	d.Set("branch", branchName)
+	d.Set("ref", *ref.Ref)
+	d.Set("sha", *ref.Object.SHA)
+
+	return nil
+}
+
+func resourceGithubBranchDelete(d *schema.ResourceData, meta interface{}) error {
+	err := checkOrganization(meta)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+
+	client := meta.(*Organization).client
+	orgName := meta.(*Organization).name
+	repoName, branchName, err := parseTwoPartID(d.Id(), "repository", "branch")
+	if err != nil {
+		return err
+	}
+	branchRefName := "refs/heads/" + branchName
+
+	log.Printf("[DEBUG] Deleting GitHub branch reference %s/%s (%s)",
+		orgName, repoName, branchRefName)
+	_, err = client.Git.DeleteRef(ctx, orgName, repoName, branchRefName)
+	if err != nil {
+		return fmt.Errorf("Error deleting GitHub branch reference %s/%s (%s): %s",
+			orgName, repoName, branchRefName, err)
+	}
+
+	return nil
+}
+
+func resourceGithubBranchImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	repoName, branchName, err := parseTwoPartID(d.Id(), "repository", "branch")
+	if err != nil {
+		return nil, err
+	}
+
+	sourceBranch := "master"
+	if strings.Contains(branchName, ":") {
+		branchName, sourceBranch, err = parseTwoPartID(branchName, "branch", "source_branch")
+		if err != nil {
+			return nil, err
+		}
+		d.SetId(buildTwoPartID(&repoName, &branchName))
```

Comment:
This line can be removed right? The resource ID won't change, you're only checking if the `source_branch` field should be set.